src/Rest/Routes/Integrations/Corvus/FormSubmitCorvusRoute.php

```diff
@@ -0,0 +1,193 @@
+		$this->labels = $labels;
+		$this->captcha = $captcha;
+		$this->security = $security;
+		$this->formSubmitMailer = $formSubmitMailer;
+	}
+
+	/**
+	 * Get the base url of the route
+	 *
+	 * @return string The base URL for route you are adding.
+	 */
+	protected function getRouteName(): string
+	{
+		return '/' . UtilsConfig::ROUTE_PREFIX_FORM_SUBMIT . '/' . self::ROUTE_SLUG;
+	}
+
+
+	/**
+	 * Implement submit action.
+	 *
+	 * @param array<string, mixed> $formDetails Data passed from the `getFormDetailsApi` function.
+	 *
+	 * @return mixed
+	 */
+	protected function submitAction(array $formDetails)
+	{
+		$formId = $formDetails[UtilsConfig::FD_FORM_ID];
+
+		if (!\apply_filters(SettingsCorvus::FILTER_SETTINGS_IS_VALID_NAME, $formId)) {
+			return \rest_ensure_response(
+				UtilsApiHelper::getApiErrorPublicOutput(
+					$this->labels->getLabel('corvusMissingConfig', $formId)
+				)
+			);
+		}
+
+		$mapParams = UtilsSettingsHelper::getSettingValueGroup(SettingsCorvus::SETTINGS_CORVUS_PARAMS_MAP_KEY, $formId);
+
+		$params = $this->prepareParams($mapParams, $formDetails['paramsRaw'], $formId);
+
+		$reqParams = [
+			'store_id',
+			'amount',
+			'language',
+			'require_complete',
+			'currency',
+			'order_number',
+			'cart',
+		];
+
+		$missingOrEmpty = \array_filter($reqParams, fn($param) => empty($params[$param] ?? null));
+
+		if ($missingOrEmpty) {
+			return \rest_ensure_response(
+				UtilsApiHelper::getApiErrorPublicOutput(
+					$this->labels->getLabel('corvusMissingReqParams', $formId)
+				)
+			);
+		}
+
+		// Add signature.
+		$params['signature'] = \hash_hmac(
+			'sha256',
+			\array_reduce(\array_keys($params), fn($carry, $key) => $carry . $key . $params[$key], ''),
+			UtilsSettingsHelper::getSettingsDisabledOutputWithDebugFilter(Variables::getApiKeyCorvus(), SettingsCorvus::SETTINGS_CORVUS_API_KEY_KEY)['value']
+		);
+
+
+		// Finish.
+		return \rest_ensure_response(
+			UtilsApiHelper::getApiSuccessPublicOutput(
+				$this->labels->getLabel('corvusSuccess', $formId),
+				[
+					UtilsHelper::getStateResponseOutputKey('postExternallyData') => [
+						'url' => $this->getUrl($formId),
+						'params' => $params,
+					],
+				]
+			)
+		);
+	}
+
+	/**
+	 * Prepare params for Corvus.
+	 *
+	 * @param array<string, string> $mapParams Map of params.
+	 * @param array<string, string> $params Form params.
+	 * @param string $formId Form ID.
+	 *
+	 * @return array<string, string>
+	 */
+	private function prepareParams(array $mapParams, array $params, string $formId): array
+	{
+		$output = [];
+
+		foreach ($mapParams as $key => $value) {
+			$param = $params[$value] ?? '';
+			if (!$param) {
+				continue;
+			}
+
+			switch ($key) {
+				case 'amount':
+					$param = \number_format((float)$param, 2, '.', '');
+					break;
+			}
+
+			$output[$key] = $param;
+		}
+
+		$output['store_id'] = UtilsSettingsHelper::getSettingValue(SettingsCorvus::SETTINGS_CORVUS_STORE_ID, $formId);
+		$output['version'] = '1.4';
```

Comment:
Reason for the hardcoded version? Please add a comment to what the version refers to. Ty

Reply:
this is the api version of the corvus API